ci/scan_flattened_deps.py

```diff
@@ -0,0 +1,343 @@
+#!/usr/bin/env python3
+#
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+#
+# Usage: scan_flattened_deps.py --flat-deps <flat DEPS file> --output <vulnerability report>
+#
+# This script parses the flattened, fully qualified dependencies,
+# and uses the OSV API to check for known vulnerabilities
+# for the given hash of the dependency
+
+import argparse
+import json
+import os
+import subprocess
+import sys
+import time
+from urllib import request
+
+SCRIPT_DIR = os.path.dirname(sys.argv[0])
+CHECKOUT_ROOT = os.path.realpath(os.path.join(SCRIPT_DIR, '..'))
+UPSTREAM_PREFIX = 'upstream_'
+
+HELP_STR = 'To find complete information on this vulnerability, navigate to '
+# TODO -- use prefix matching for this rather than always to OSV
```

Comment:
TODOs should indicate a person and a GitHub issue. E.g.
`// TODO(sealesj): Use prefix matching: <github issue URL>`